webapp/src/ts/modules/contacts/contacts-edit.component.ts

```diff
@@ -184,46 +190,36 @@ export class ContactsEditComponent implements OnInit, OnDestroy, AfterViewInit {
       .then((result) => result.doc);
   }
 
-  private getForm(contact) {
+  private getForm(contact, contactType) {
     let formId;
     let titleKey;
-    const typeId = this.contactTypesService.getTypeId(contact) || this.routeSnapshot.params?.type;
-    return this.contactTypesService
-      .get(typeId)
-      .then(type => {
-        if (!type) {
-          console.error(`Unknown contact type "${typeId}"`);
-          return;
-        }
+    if (contact) { // editing
+      this.contact = contact;
+      this.contactId = contact._id;
+      titleKey = contactType.edit_key;
+      formId = contactType.edit_form || contactType.create_form;
+    } else { // adding
+      this.contact = {
+        type: 'contact',
+        contact_type: this.routeSnapshot.params?.type,
+        parent: this.routeSnapshot.params?.parent_id || '',
+      };
+      this.contactId = null;
+      formId = contactType.create_form;
+      titleKey = contactType.create_key;
+    }
 
-        if (contact) { // editing
-          this.contact = contact;
-          this.contactId = contact._id;
-          titleKey = type.edit_key;
-          formId = type.edit_form || type.create_form;
-        } else { // adding
-          this.contact = {
-            type: 'contact',
-            contact_type: this.routeSnapshot.params?.type,
-            parent: this.routeSnapshot.params?.parent_id || '',
-          };
-          this.contactId = null;
-          formId = type.create_form;
-          titleKey = type.create_key;
+    this.translationsLoadedSubscription?.unsubscribe();
+    this.translationsLoadedSubscription = this.store
+      .select(Selectors.getTranslationsLoaded)
+      .subscribe((loaded) => {
+        if (loaded) {
+          this.translateService
+            .get(titleKey)
```

Comment:
There is a little duplication here with setting the page title, where both the page title and the form title get the value from the contactType details. Maybe we should de-duplicate this as well, and reuse the title for the action AND the enketo html title.